src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -824,50 +908,27 @@ def get_stack(self, stack_id: UUID) -> StackResponseModel:
             return stack.to_model()
 
     def list_stacks(
-        self,
-        project_name_or_id: Optional[Union[str, UUID]] = None,
-        user_name_or_id: Optional[Union[str, UUID]] = None,
-        component_id: Optional[UUID] = None,
-        name: Optional[str] = None,
-        is_shared: Optional[bool] = None,
-    ) -> List[StackResponseModel]:
+        self, stack_filter_model: StackFilterModel
+    ) -> Page[StackResponseModel]:
         """List all stacks matching the given filter criteria.
 
         Args:
-            project_name_or_id: ID or name of the Project containing the stack
-            user_name_or_id: Optionally filter stacks by their owner
-            component_id: Optionally filter for stacks that contain the
-                          component
-            name: Optionally filter stacks by their name
-            is_shared: Optionally filter out stacks by whether they are shared
-                or not
+            stack_filter_model: All filter parameters including pagination
+                                params
 
         Returns:
             A list of all stacks matching the filter criteria.
         """
         with Session(self.engine) as session:
-            # Get a list of all stacks
+            # Manually create the query and add any custom clauses
```

Comment:
I would remove this comment, after your changes all the list commands feel very clean and at least to me it is obvious what they do / how they work now. Same for other list methods below

```suggestion
```

Reply:
All removed already